src/System.CommandLine/Binding/ModelBinder.cs

```diff
@@ -33,9 +33,29 @@ internal ModelBinder(IValueDescriptor valueDescriptor)
 
         public void BindMemberFromValue(PropertyInfo property, IValueDescriptor valueDescriptor)
         {
-            NamedValueSources.Add(
-                (property.PropertyType, property.Name),
-                new SpecificSymbolValueSource(valueDescriptor));
+            var key = (property.PropertyType, property.Name);
+            if (NamedValueSources.TryGetValue(key, out var existingValueSource) &&
+                existingValueSource is null)
```

Comment:
Rather than assign a semantic meaning to null and then always have to check for null, a type representing a strict binding might be preferable. Although another approach that doesn't require populating the dictionary might make things easier to understand.